src/Illuminate/Database/Grammar.php

```diff
@@ -116,6 +116,62 @@ protected function wrapValue($value)
         return $value;
     }
 
+    /**
+     * Convert array into sql group.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function group($values)
+    {
+        return '('.implode(', ', $values).')';
+    }
+
+    /**
+     * Recursively group arrays containing values into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupify($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
```

Comment:
Is there any valid sql query where recursion is required? I would probably just map the `parameter` method to it.

Reply:
Since you are sending in an array of arrays this one method will for the most part recurse once before converting all values and then bubbling back up grouping the arrays first together before grouping the entire result